201-api-management-create-with-internal-vnet-application-gateway/azuredeploy.json

```diff
@@ -0,0 +1,511 @@
+                    "name": "applicationinsights",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.ApiManagement/service', variables('apim-name'))]",
+                        "[resourceId('Microsoft.ApiManagement/service/loggers', variables('apim-name'), 'AppInsightsLogger')]"
+                    ],
+                    "properties": {
+                        "alwaysLog": "allErrors",
+                        "httpCorrelationProtocol": "Legacy",
+                        "verbosity": "information",
+                        "logClientIp": true,
+                        "loggerId": "[resourceId('Microsoft.ApiManagement/service/loggers/', variables('apim-name'), 'AppInsightsLogger')]",
+                        "sampling": {
+                            "samplingType": "fixed",
+                            "percentage": 100
+                        },
+                        "frontend": {
+                            "request": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            }
+                        },
+                        "backend": {
+                            "request": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "bytes": 0
+                                }
+                            }
+                        }
+                    }
+                },
+                {
+                    "type": "providers/diagnosticSettings",
+                    "apiVersion": "2017-05-01-preview",
+                    "name": "Microsoft.Insights/logToAnalytics",
+                    "location": "[parameters('location')]",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.ApiManagement/service', variables('apim-name'))]"
+                    ],
+                    "properties": {
+                        "workspaceId": "[parameters('log-analytics-workspace-id')]",
+                        "logs": [
+                            {
+                                "category": "GatewayLogs",
+                                "enabled": true
+                            }
+                        ],
+                        "metrics": [
+                            {
+                                "category": "AllMetrics",
+                                "enabled": true
+                            }
+                        ]
+                    }
+                }
+            ],
+            "dependsOn": [
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

Reply:
Done - didn't catch "resources" as falling in the "everything not on this list goes before 'properties'"